WeightProducer/python/MCSampleValues.py

```diff
@@ -63,493 +63,1039 @@ class MCSampleValuesHelper():
+                XS_2017=57.48, XSSource_2017="XSDB (LO)",
+                XS_2018=57.80, XSSource_2018="GenXSecAnalyzer",
+            ),
+            "kFactor" : kFactorValues(
+                kFactor_2016=1.224, kFactorSource_2016="XSDB NNLO/LO=61526.7/50260",
+                kFactor_2017=1.162, kFactorSource_2017="XSDB NNLO/LO=61526.7/52940",
+                kFactor_2018=1.164, kFactorSource_2018="XSDB NNLO/LO=61526.7/52850",
+            ),
         },
         "WJetsToLNu_HT-600To800" : {
-            "CrossSection" : XSValues(XS_13TeV=14.58),
-            "kFactor" : kFactorValues(kFactor_13TeV=1.21),
+            "CrossSection" : XSValues(
+                XS_2016=12.07, XSSource_2016="XSDB (LO)",
+                XS_2017=12.87, XSSource_2017="XSDB (LO)",
+                XS_2018=12.94, XSSource_2018="GenXSecAnalyzer",
+            ),
+            "kFactor" : kFactorValues(
+                kFactor_2016=1.224, kFactorSource_2016="XSDB NNLO/LO=61526.7/50260",
+                kFactor_2017=1.162, kFactorSource_2017="XSDB NNLO/LO=61526.7/52940",
+                kFactor_2018=1.164, kFactorSource_2018="XSDB NNLO/LO=61526.7/52850",
+            ),
         },
         "WJetsToLNu_HT-800To1200" : {
-            "CrossSection" : XSValues(XS_13TeV=6.66),
-            "kFactor" : kFactorValues(kFactor_13TeV=1.21),
+            "CrossSection" : XSValues(
+                XS_2016=5.497, XSSource_2016="XSDB (LO)",
+                XS_2017=5.366, XSSource_2017="XSDB (LO)",
+                XS_2018=5.451, XSSource_2018="GenXSecAnalyzer",
+            ),
+            "kFactor" : kFactorValues(
+                kFactor_2016=1.224, kFactorSource_2016="XSDB NNLO/LO=61526.7/50260",
+                kFactor_2017=1.162, kFactorSource_2017="XSDB NNLO/LO=61526.7/52940",
+                kFactor_2018=1.164, kFactorSource_2018="XSDB NNLO/LO=61526.7/52850",
+            ),
         },
         "WJetsToLNu_HT-1200To2500" : {
-            "CrossSection" : XSValues(XS_13TeV=1.608),
-            "kFactor" : kFactorValues(kFactor_13TeV=1.21),
+            "CrossSection" : XSValues(
+                XS_2016=1.329, XSSource_2016="XSDB (LO)",
+                XS_2017=1.074, XSSource_2017="XSDB (LO)",
+                XS_2018=1.085, XSSource_2018="GenXSecAnalyzer",
+            ),
+            "kFactor" : kFactorValues(
+                kFactor_2016=1.224, kFactorSource_2016="XSDB NNLO/LO=61526.7/50260",
+                kFactor_2017=1.162, kFactorSource_2017="XSDB NNLO/LO=61526.7/52940",
+                kFactor_2018=1.164, kFactorSource_2018="XSDB NNLO/LO=61526.7/52850",
+            ),
         },
         "WJetsToLNu_HT-2500ToInf" : {
-            "CrossSection" : XSValues(XS_13TeV=0.03891),
-            "kFactor" : kFactorValues(kFactor_13TeV=1.21),
+            "CrossSection" : XSValues(
+                XS_2016=0.03209, XSSource_2016="XSDB (LO)",
+                XS_2017=0.008001, XSSource_2017="XSDB (LO)",
```

Comment:
this is a 4x change from 2016 -> 2017

Reply:
I'll run the 2016 sample through the GenXSecAnalyzer and see if it matches the value there or if it's closer to the 2017/2018 values.

Reply:
I've run the GenXSecAnalyzer for the 2016 sample and it confirms the cross section found in the XSDB. I'm still not sure why this cross section changed so dramatically. I'll ask some GEN contacts.

Reply:
This seems to be related to the second issue (HT stitching) reported at https://twiki.cern.ch/twiki/bin/viewauth/CMS/MCKnownIssues#WJetsToLNu_HT_and_DYJets_HT_LO_M. There they recommend applying a set of "per-bin cross section correction factors ... derived based on comparing the HT distribution between the HT-binned and inclusive samples." I recommend we add another namedtuple_with_defaults type (i.e. correction_factors), where the default value is 1. Then we can scale the cross sections as recommended. This will maintain the provenance of the corrections and allow for maximum flexibility.

Reply:
I think this is a good idea

Reply:
This should be implemented now.